monkey/infection_monkey/control.py

```diff
@@ -78,22 +77,12 @@ def send_log(self, log):
             return
         try:
             telemetry = {"monkey_guid": GUID, "log": json.dumps(log)}
-            requests.post(  # noqa: DUO123
-                "https://%s/api/log" % (self.server_address,),
-                data=json.dumps(telemetry),
-                headers={"content-type": "application/json"},
-                verify=False,
-                timeout=MEDIUM_REQUEST_TIMEOUT,
-            )
+            HTTPIslandAPIClient.send_log(self.server_address, json.dumps(telemetry))
         except Exception as exc:
             logger.warning(f"Error connecting to control server {self.server_address}: {exc}")
 
     def get_pba_file(self, filename):
         try:
-            return requests.get(  # noqa: DUO123
-                PBA_FILE_DOWNLOAD % (self.server_address, filename),
-                verify=False,
-                timeout=LONG_REQUEST_TIMEOUT,
-            )
-        except requests.exceptions.RequestException:
-            return False
+            HTTPIslandAPIClient.get_pba_file(self.server_address, filename)
```

Comment:
`ControlClient` will need to take an `IIslandAPIClient` in its constructor. Everything should relay on the interface, not the concrete implementations.